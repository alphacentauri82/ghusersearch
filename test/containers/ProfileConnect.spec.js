import {
  mapDispatchToProps,
  mapStateToProps,
} from 'containers/Profile/connect';
import {isFSA} from 'flux-standard-action';

describe('ProfileConnect: mapDispatchToProps', () => {

  describe('when dispatching an action', () => {
    it('should conform to FSA', () => {
      const fakeDispatch = jest.fn();
      mapDispatchToProps(fakeDispatch).getProfile('simonsmith');
      expect(isFSA(fakeDispatch.mock.calls[0][0])).toBeTruthy();
    });

    it('should ignore undefined arguments', () => {
      const fakeDispatch = jest.fn();
      expect(mapDispatchToProps(fakeDispatch).getProfile()).toBeUndefined();
    });
  });

});

describe('ProfileConnect: mapStateToProps', () => {

  describe('when given the state', () => {
    it('should return relevant properties', () => {
      const state = {
        repos: {
          result: [1, 2, 3],
          isPending: false,
        },
        followers: {
          result: [4, 5, 6],
          isPending: false,
        },
        profile: {
          userProfile: {},
          isPending: false,
        },
        entities: {
          repos: {
            1: {id: 1},
            2: {id: 2},
          },
          users: {
            4: {id: 4},
            5: {id: 5},
          },
        },
      };
      expect(mapStateToProps(state)).toMatchSnapshot();
    });
  });

});
