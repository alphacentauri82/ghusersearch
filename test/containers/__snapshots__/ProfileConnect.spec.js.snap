// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileConnect: mapStateToProps when given the state should return relevant properties 1`] = `
Object {
  "followerIds": Array [
    4,
    5,
    6,
  ],
  "followerIsPending": false,
  "repoEntities": Object {
    "1": Object {
      "id": 1,
    },
    "2": Object {
      "id": 2,
    },
  },
  "repoIds": Array [
    1,
    2,
    3,
  ],
  "repoIsPending": false,
  "userEntities": Object {
    "4": Object {
      "id": 4,
    },
    "5": Object {
      "id": 5,
    },
  },
  "userIsPending": false,
  "userProfile": Object {},
}
`;
